src/Caching/Abstractions/ref/Microsoft.Extensions.Caching.Abstractions.csproj

```diff
@@ -1,10 +1,14 @@
 <!-- This file is automatically generated. -->
 <Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
-    <TargetFrameworks>netstandard2.0</TargetFrameworks>
+    <TargetFrameworks>netstandard2.0;netcoreapp3.0</TargetFrameworks>
   </PropertyGroup>
   <ItemGroup Condition="'$(TargetFramework)' == 'netstandard2.0'">
     <Compile Include="Microsoft.Extensions.Caching.Abstractions.netstandard2.0.cs" />
     <Reference Include="Microsoft.Extensions.Primitives"  />
   </ItemGroup>
+<ItemGroup Condition="'$(TargetFramework)' == 'netcoreapp3.0'">
```

Comment:
Why add a netcoreapp3.0 target in projects that reference exactly the same packages without that TFM?
  - I could see this if all ref/ projects use this approach while all non-ref/ projects use `$(DotNetBuildFromSource)`. But, don't we also need to source build the ref/ projects _only_ for netcoreapp3.0? 
  - Put another way, when building for source build, what will narrow the list of target frameworks in ref/ projects to just netcoreapp3.0?
  - Note I didn't confirm all ref/ projects are unconditionally multitargeted

Can you or GenAPI reduce the duplication by removing the conditions and using e.g. `<Compile Include="Microsoft.Extensions.Caching.Abstractions.$(TargetFramework).cs" />`?
  - Overall condition of the single reference / compile list could just be ` '$(TargetFramework)' != '' `

Separately, might help to update the issues to explicitly say we're going with netcoreapp3.0 and not netstandard2.1

Reply:
To my understanding, the ref/ projects are not built in source-build: https://github.com/dotnet/source-build/pull/503. @crummel @dagood is that still accurate?

Reply:
@JunTaoLuo That PR was about removing some source-build infra that was producing some  ref assembly packages in a way that didn't work very well. Now we have a new way of producing ref assembly packages that works better. But `ref/` projects are unrelated--they've always built in any repo that has the `ref/` `src/` pattern to my knowledge.

Reply:
Ah, I misunderstood the PR. Hmm I need to think about this more. I regenerated the reference assemblies without considering the side effects.

One way to do this in source build might be to rerun our GenerateReferenceAssemblies script in source build.

Reply:
Oh and w.r.t 
> Why add a netcoreapp3.0 target in projects that reference exactly the same packages without that TFM?

We need to multi-target projects that have multi-targeted dependencies, otherwise the SDK will complain:
```
C:\gh\Extensions\.dotnet\sdk\3.0.100-preview7-012821\Microsoft.Common.CurrentVersion.targets(1653,5): error : Project 'C:\gh\Extensions\src\Primitives\src\Microsoft.Extensions.Primitives.csproj' targets 'netcoreapp3.0'. It cannot be referenced by a project that targets '.NETStandard,Version=v2.0'. [C:\gh\Extensions\src\Caching\Abstractions\src\Microsoft.Extensions.Caching.Abstractions.csproj]
```

For example, the above error will show up if you multi-target M.E.Primitives with netstandard2.0 and netcoreapp3.0 but keep M.E.Caching.Abstractions only at netstandard2.0.